python/cmsis_svd/model.py

```diff
@@ -1,452 +1,620 @@
                     pubkey = k[1:]
-                    eldict[pubkey] = getattr(obj, pubkey)
+                    eldict[pubkey] = getattr(o, pubkey)
                 else:
                     eldict[k] = v
             return eldict
-        else:
-            return json.JSONEncoder.default(self, obj)
 
+        if isinstance(o, Enum):
+            return o.value
 
-class SVDElement(object):
-    """Base class for all SVD Elements"""
+        return json.JSONEncoder.default(self, o)
 
-    def __init__(self):
-        self.parent = None
 
-    def to_dict(self):
+@dataclass
+class SVDElement:
+    """Base class for all SVD Elements"""
+    parent: Optional['SVDElement'] = field(default=None)
+
+    def to_dict(self) -> Any:
         # This is a little convoluted but it works and ensures a
         # json-compatible dictionary representation (at the cost of
         # some computational overhead)
         encoder = SVDJSONEncoder()
         return json.loads(encoder.encode(self))
 
 
+################################################################################
+# CMSIS-SVD Model Type
+################################################################################
+
+class SVDAccessType(Enum):
+    READ_ONLY = 'read-only'
+    WRITE_ONLY = 'write-only'
+    READ_WRITE = 'read-write'
+    WRITE_ONCE = 'writeOnce'
+    READ_WRITE_ONCE = 'read-writeOnce'
+
+
+class SVDAddressBlockUsageType(Enum):
+    REGISTERS = 'registers'
+    BUFFER = 'buffer'
+    RESERVED = 'reserved'
+
+
+class SVDDataTypeType(Enum):
+    UINT8_T = 'uint8_t'
+    UINT16_T = 'uint16_t'
+    UINT32_T = 'uint32_t'
+    UINT64_T = 'uint64_t'
+    INT8_T = 'int8_t'
+    INT16_T = 'int16_t'
+    INT32_T = 'int32_t'
+    INT64_T = 'int64_t'
+    UINT8_T_1 = 'uint8_t *'
+    UINT16_T_1 = 'uint16_t *'
+    UINT32_T_1 = 'uint32_t *'
+    UINT64_T_1 = 'uint64_t *'
+    INT8_T_1 = 'int8_t *'
+    INT16_T_1 = 'int16_t *'
+    INT32_T_1 = 'int32_t *'
+    INT64_T_1 = 'int64_t *'
+
+
+class SVDEndianType(Enum):
+    LITTLE = 'little'
+    BIG = 'big'
+    SELECTABLE = 'selectable'
+    OTHER = 'other'
+
+
+class SVDEnumUsageType(Enum):
+    READ = "read"
+    WRITE = "write"
+    READ_WRITE = "read-write"
+
+
+class SVDModifiedWriteValuesType(Enum):
+    ONE_TO_CLEAR = 'oneToClear'
+    ONE_TO_SET = 'oneToSet'
+    ONE_TO_TOGGLE = 'oneToToggle'
+    ZERO_TO_CLEAR = 'zeroToClear'
+    ZERO_TO_SET = 'zeroToSet'
+    ZERO_TO_TOGGLE = 'zeroToToggle'
+    CLEAR = 'clear'
+    SET = 'set'
+    MODIFY = 'modify'
+
+
+class SVDReadActionType(Enum):
+    CLEAR = 'clear'
+    SET = 'set'
+    MODIFY = 'modify'
+    MODIFY_EXTERNAL = 'modifyExternal'
+
+
+SvdRegistersListType = List[Union[
+    'SVDRegister', 'SVDRegisterCluster', 'SVDRegisterArray',
+    'SVDRegisterClusterArray'
+]]
+
+################################################################################
+# CMSIS-SVD Model Special Elements
+################################################################################
+
+
```

Comment:
nit: Only one blank line here, or other headers should use 2 blanks afterwards.

Reply:
I will push a fix.

Reply:
Fixed